### ISSUE TITLE
connected/disconnected event emiter not properly exported

### ISSUE THREAD

**Issue by glend1** [djipco/webmidi#226]
thought i had a issue open for this, but i can't seem to find it so im reposting. sorry if im just being blind.

when I do the following

````
WebMidi.addListener("connected", (e) => { console.log(e.target.name)});
````

i get the following typescript error

````
Property 'name' does not exist on type 'Input | WebMidi | InputChannel | Output'.
    Property 'name' does not exist on type 'WebMidi'.ts(2339)
````

**Comment by djipco** [comment]
I just tried the code below in Node.js and I'm not getting errors with v3.0.10:

```typescript
const {WebMidi} = require("webmidi");
WebMidi.addListener("connected", e => console.log(e.target.name));
WebMidi.enable();
```

Can you provide a little more info about the environment where you get that error?

**Comment by glend1** [comment]
I am using Next.js which doesn't like using `require` so I converted your code to an `import` (you are able to bypass as shown in the following code snippet with `export {}`)

````
//import {WebMidi} from "webmidi";
const {WebMidi} = require("webmidi");
WebMidi.addListener("connected", e => console.log(e.target.name));
WebMidi.enable();

export{}
````

however, with `require` I notice that `e` is `any`.

**Comment by djipco** [comment]
There must be something different in your environment because it works fine for me. Your code example ran fine in a demo Next.js project. I tried with `require` and `import` and both worked fine.

**Comment by djipco** [comment]
Hold on... I think I found the issue. Let me get back to you.

**Comment by djipco** [comment]
I took a deeper look at events and concluded that all events that are triggered by a port (`Input` or `Output`) needed their own type definitions for TypeScript. So, I have added a `PortEvent` type that extends the basic `Event` type. The `PortEvent` type has a `port` property that points to the ` Input` of `Output` port. 

So, in you case, you should be able to use `e.port.name` just like before.

This will also help when adding a listener to an `InputChannel`. In this case, `e.port` will point to the `Input` object while `e.target` will point to the `InputChannel` object.

This change is available right now in version 3.0.11.
